account_statement_import_txt_xlsx/models/account_statement_import_sheet_parser.py

```diff
@@ -0,0 +1,389 @@
+                ).copy_sign(amount)
+            else:
+                original_amount = 0.0
+
+            line = {
+                "timestamp": timestamp,
+                "amount": amount,
+                "currency": currency,
+                "original_amount": original_amount,
+                "original_currency": original_currency,
+            }
+            if balance is not None:
+                line["balance"] = balance
+            if transaction_id is not None:
+                line["transaction_id"] = transaction_id
+            if description is not None:
+                line["description"] = description
+            if notes is not None:
+                line["notes"] = notes
+            if reference is not None:
+                line["reference"] = reference
+            if partner_name is not None:
+                line["partner_name"] = partner_name
+            if bank_name is not None:
+                line["bank_name"] = bank_name
+            if bank_account is not None:
+                line["bank_account"] = bank_account
+            lines.append(line)
+        return lines
+
+    @api.model
+    def _convert_line_to_transactions(self, line):  # noqa: C901
+        """Hook for extension"""
+        timestamp = line["timestamp"]
+        amount = line["amount"]
+        currency = line["currency"]
+        original_amount = line["original_amount"]
+        original_currency = line["original_currency"]
+        transaction_id = line.get("transaction_id")
+        description = line.get("description")
+        notes = line.get("notes")
+        reference = line.get("reference")
+        partner_name = line.get("partner_name")
+        bank_name = line.get("bank_name")
+        bank_account = line.get("bank_account")
+
+        transaction = {
+            "date": timestamp,
+            "amount": str(amount),
+        }
+
+        if original_currency == currency:
+            original_currency = None
+            if not amount:
+                amount = original_amount
+            original_amount = "0.0"
+
+        if original_currency:
+            original_currency = self.env["res.currency"].search(
+                [("name", "=", original_currency)],
+                limit=1,
+            )
+            if original_currency:
+                transaction["foreign_currency_id"] = original_currency.id
+            if original_amount:
+                transaction["amount_currency"] = str(original_amount)
+
+        if currency:
+            currency = self.env["res.currency"].search(
+                [("name", "=", currency)],
+                limit=1,
+            )
+            if currency:
+                transaction["currency_id"] = currency.id
+
+        if transaction_id:
+            transaction["unique_import_id"] = "{}-{}".format(
+                transaction_id,
+                int(timestamp.timestamp()),
+            )
+
+        transaction["name"] = description or _("N/A")
```

Comment:
```suggestion
        transaction["payment_ref"] = description or _("N/A")
```
Now in v14 the model `account.bank.statement.line` inherits from `account.move` (composition-based inheritance) so the name field is the same as defined in the `account.move` model. In other words, it will be the number of the related journal entry, so we shouldn't change it. However, the column that is now displayed with the name 'Label' is the one corresponding to the `payment_ref` field.

Reply:
Well in my tests I've found that the first `account.bank.statement.line` does get a meaningful name/number like BNK1/2021/.../..., the following ones don't. If the name field isn't displayed in the most used views, I agree to leave it alone, even if I'd like the records created to have proper names.

Anyway, I think we agree here. I decided to simply not pass the name at all:
```      # avoid errors due to sequence checking
        del transaction["name"]
```
since transaction is a dictionary that is (cutting a long story short) used as values in a create().  

The final point (where the sequence checking happens and the error is triggered) is here:

https://github.com/OCA/OCB/blob/c0a79cdda56bf0aa4ee6741dce7eaff0179f857b/addons/account/models/account_bank_statement.py#L859

I've tried to simply disable the sequence mechanism, by overriding the mixin methods in `account.bank.statement.line`, that doesn't seem to work, in that call is seems the methods in `account.move` are called directly (I've traced it). I've stopped investigating further because I'm not sure if  `account.bank.statement.line` is inheriting from `account.move` just for convenience (to avoid code repetition) or those records are actually and fully intended to be `account.move`'s (and I mean by standard accounting practice).  To put it differently, if it was a python programmer's idea or an accountant's idea.  If the second is true, and they are supposed to be real `account.move`'s, probably they _should_ have proper sequence numbers as names and disabling the checks isn't the appropriate solution.


All this to say I like your approach more. In short:
- don't create the key `name` in the first place, at all
- use `payment_ref` instead (and put just the `description_column` in it)
- use `narration` for notes


BTW the "long story" I've mentioned before goes like this:

the _parse_file() method in `account.statement.import` (which is an abstract method you should override is sub-modules) returns a whose data structures, in particulare the transactions list is what we're interested in:

https://github.com/OCA/bank-statement-import/blob/98618c06117f9f6ebdecd361e741897d79393523/account_statement_import/wizard/account_statement_import.py#L135-L163
note that the comment mentions `payment_ref` and not `name`.

The transactions undergo to some checking and sanitation, the finally they are turned into `line_ids` here:
https://github.com/OCA/bank-statement-import/blob/98618c06117f9f6ebdecd361e741897d79393523/account_statement_import/wizard/account_statement_import.py#L354-L358

So my assessment was, we either provide meaningful names or don't even have the key in our dictionaries. I was unable to find a way to provide a meaningful name, so I opted for the second route, and I decided to remove the key at the end. Like I said, I agree it's best not to create it in the first place.

Reply:
Thanks to answer @TheMule71. I really do not know if taking the lines of the bank statements as journal entries is the best idea or not, but we have no choice to deal with it, hehe.

Anyway `del transaction["name"]` throws an error, because the key 'name' is not in the dictionay `transaction`, that's why I just propused to change `transaction["name"]` to `transaction["payment_ref"]`.

ping @primes2h

Reply:
> Now in v14 the model `account.bank.statement.line` inherits from `account.move` (composition-based inheritance) so the name field is the same as defined in the `account.move` model. In other words, it will be the number of the related journal entry, so we shouldn't change it. However, the column that is now displayed with the name 'Label' is the one corresponding to the `payment_ref` field.

Done, thanks!

Reply:
@TheMule71
Name/Sequence is not an issue anymore. 
Do you think it still worth adding [this test](https://github.com/PNLUG/bank-statement-import/pull/5/commits/a0fb268a99f9dd04a904553d968e7d4b663b9443#diff-3c2f9c9f9192ba5e4d53d38654893c14ae6596292af17e96967742a2b9f8a2df)? Otherwise https://github.com/PNLUG/bank-statement-import/pull/5/ can be closed.

Reply:
No, I think it's not necessary to add that test, because the only thing that it tested (that has not been tested in other tests) is the issue with Name/Sequence and we no longer have it.
ping @TheMule71

Reply:
I agree that having a second import_csv test adds nothing and can be removed, but I'd still put a line with a number in the description as part of the regular test_import_csv_file() test we already have.

The cost is negligible and we're covering a case that did prove problematic in the past.

Closing https://github.com/PNLUG/bank-statement-import/pull/5

Reply:
> I agree that having a second import_csv test adds nothing and can be removed, but I'd still put a line with a number in the description as part of the regular test_import_csv_file() test we already have.

Done, thank you!